### DISCUSSION TITLE
兼容@import '~es/style/index.less'; 引入less 文件

### DISCUSSION THREAD

**Discussion by wzlwangzhilin** [umijs/umi#12040] [Q&A]
![20240109-005209](https://github.com/umijs/umi/assets/91478474/d8ca1587-d834-4f98-8739-9237c66a47b9)
编译时报错

**Comment by fz6m** [comment]
确认下你的依赖里是否有这个文件，尝试升级下 umi 版本到最新，如还有问题，需要给一个最小复现。

我使用 arco design 是没问题的，在 `src/global.ts` 这个全局脚本文件里导入 arco 全局样式：

```ts
// src/global.ts
import "@arco-design/web-react/dist/css/arco.css";
// or
// import "@arco-design/web-react/dist/css/index.less";
```

或者在全局样式文件 `src/global.less` 里导入：

```ts
// src/global.less
@import '@arco-design/web-react/dist/css/index.less';
```

或者导入：

```ts
// src/global.less
@import '~@arco-design/web-react/es/style/theme/default.less';
// or @import '@arco-design/web-react/es/style/theme/default.less';
```

都是正常的。

**Reply by wzlwangzhilin** [reply]
![image](https://github.com/umijs/umi/assets/91478474/2fb8b8c3-8a90-4084-b472-651637a3bed4)
![image](https://github.com/umijs/umi/assets/91478474/e3d7ff06-673e-4539-b701-b28ed53e4096)
![image](https://github.com/umijs/umi/assets/91478474/a561d371-a5af-4d93-8124-b9371f44aeab)
这么用就能简单复现。我理解是less文件里～引入的问题。但是是第三方包里的 我没法修改
![image](https://github.com/umijs/umi/assets/91478474/614dab0d-d4e3-445c-a9a3-7962995f41de)
我理解应该是这个问题。但我没找着在umi里应该怎么解决
![image](https://github.com/umijs/umi/assets/91478474/9eb17049-be8a-4047-a08d-96fd84c73a8a)
公司内部其他工具配置是 但我
![image](https://github.com/umijs/umi/assets/91478474/527111cd-0215-4ed3-b6e2-92f4d3fb6ed3)
这么配没用。不知道有没有其他解决办法

**Comment by wzlwangzhilin** [comment]


**Reply by wzlwangzhilin** [reply]
![image](https://github.com/umijs/umi/assets/91478474/9be48f73-699a-4386-ba8a-58acbdf8a25b)
我理解应该是这个问题。但我没找着在umi里应该怎么解决

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4 。
